### PR TITLE
[WIP] feat(asyncapi): make asyncapi-payload-* rules support asyncapi schemaFormats 

### PR DESCRIPTION
**Pull request by nulltoken**
**Checklist**

- [x] Tests added / updated
- [x] Docs added / updated

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

Follow up work on #974 AsyncAPI PR.

- Fixes a technical debt I introduced
- Paves the way to make it easier to validate payloads against other schemas (Avro, OAS3, ...)

### CODE REVIEW

**Review by philsturgeon — DISMISSED** [review]

**Review by fmvilas — DISMISSED** [review]
LGTM 👍

File: src/rulesets/asyncapi/functions/__tests__/asyncApi2PayloadValidation.test.ts [hunk 1]
```diff
@@ -1,30 +1,68 @@
...
 }
 
 describe('asyncApi2PayloadValidation', () => {
-  test('Properly identify payload that do not fit the AsyncApi2 schema object definition', () => {
-    const payload = {
-      type: 'object',
-      deprecated: 14,
-    };
+  describe('AsyncApi2 schema', () => {
+    const validSchemaFormats = [
```
Comment (nulltoken) [inline src/rulesets/asyncapi/functions/__tests__/asyncApi2PayloadValidation.test.ts:17]: @nulltoken Use exported values from the func

**Comment by P0lip** [comment]
Going to close this one.
If you ever happen to find some time to get this one over the finish line, do feel free to reopen it.
If you still remember, you can also tell me what's left here and I may get it done in my spare time.

**Pull request closed without merging**